cats/__init__.py

```diff
@@ -23,17 +26,65 @@ def parse_arguments():
 
     #parser.add_argument("program")
     parser.add_argument("--loc")
-    parser.add_argument("-d", "--duration", required=True)
+    parser.add_argument("-d", "--duration", type=int, required=True)
+    parser.add_argument("--jobinfo")
+    parser.add_argument("--config")
 
     return parser
 
 
+def validate_jobinfo(jobinfo: str):
+    """Parses a string of job info keys in the form
+
+    partition=CPU_partition,memory=8,ncpus=8,ngpus=0
+
+    and checks all required info keys are present and of the right type.
+
+    Returns
+    -------
+
+    info: dict
+        A dictionary mapping info key to their specified values
+    """
+
+    expected_info_keys = (
+        "partition",
+        "memory",
+        "cpus",
+        "gpus",
+    )
+    info = {
+        k: v
+        for k, v in [match.groups() for match in re.finditer(r"(\w+)=(\w+)", jobinfo)]
+    }
+    missing_keys = set(expected_info_keys) - set(info.keys())
+    if missing_keys:
+        print(f"ERROR: Missing job info keys: {missing_keys}")
+        return {}
+    expected_partition_values = ("CPU_partition", "GPU_partition")
+    if info["partition"] not in expected_info_keys:
```

Comment:
Typo right there: `expected_info_keys` -> `expected_partition_values`

Reply:
haha yes was about to comment on this!

Reply:
thanks for fixing!